File: flutter_app/lib/data/models/student.dart
class Student {
  final int id;
  final int classId;
  final String name;
  final String gender;
  final String? phone;
  final String? parentPhone;
  final String? remarks;
  final DateTime createdAt;

  // Extended fields
  final String? studentNo;
  final String? birthday;
  final String? parentName;
  final String? address;
  final String? homePhone;
  final String? hobbies;
  final String? health;
  final String? emergencyContact;
  final String? description;

  Student({
    required this.id,
    required this.classId,
    required this.name,
    required this.gender,
    this.phone,
    this.parentPhone,
    this.remarks,
    required this.createdAt,
    this.studentNo,
    this.birthday,
    this.parentName,
    this.address,
    this.homePhone,
    this.hobbies,
    this.health,
    this.emergencyContact,
    this.description,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'],
      classId: json['class_id'],
      name: json['name'],
      gender: json['gender'],
      phone: json['phone'],
      parentPhone: json['parent_phone'],
      remarks: json['remarks'],
      createdAt: DateTime.parse(json['created_at']),
      studentNo: json['student_no'],
      birthday: json['birthday'],
      parentName: json['parent_name'],
      address: json['address'],
      homePhone: json['home_phone'],
      hobbies: json['hobbies'],
      health: json['health'],
      emergencyContact: json['emergency_contact'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'gender': gender,
      if (phone != null) 'phone': phone,
      if (parentPhone != null) 'parent_phone': parentPhone,
      if (remarks != null) 'remarks': remarks,
      if (studentNo != null) 'student_no': studentNo,
      if (birthday != null) 'birthday': birthday,
      if (parentName != null) 'parent_name': parentName,
      if (address != null) 'address': address,
      if (homePhone != null) 'home_phone': homePhone,
      if (hobbies != null) 'hobbies': hobbies,
      if (health != null) 'health': health,
      if (emergencyContact != null) 'emergency_contact': emergencyContact,
      if (description != null) 'description': description,
    };
  }
}

File: backend/app/models/student.py
from datetime import datetime, date, UTC
from sqlalchemy import String, DateTime, Date, ForeignKey, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database import Base


class Student(Base):
    __tablename__ = "students"

    id: Mapped[int] = mapped_column(primary_key=True)
    class_id: Mapped[int] = mapped_column(ForeignKey("classes.id"))
    name: Mapped[str] = mapped_column(String(100))
    gender: Mapped[str] = mapped_column(String(10))  # male, female
    phone: Mapped[str | None] = mapped_column(String(20), nullable=True)
    parent_phone: Mapped[str | None] = mapped_column(String(20), nullable=True)
    remarks: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(UTC))

    # ── Extended fields (all optional) ──────────────────────────────────────
    student_no: Mapped[str | None] = mapped_column(String(50), nullable=True)
    birthday: Mapped[date | None] = mapped_column(Date, nullable=True)
    parent_name: Mapped[str | None] = mapped_column(String(100), nullable=True)
    address: Mapped[str | None] = mapped_column(String(300), nullable=True)
    home_phone: Mapped[str | None] = mapped_column(String(20), nullable=True)
    hobbies: Mapped[str | None] = mapped_column(String(300), nullable=True)
    health: Mapped[str | None] = mapped_column(String(300), nullable=True)
    emergency_contact: Mapped[str | None] = mapped_column(String(100), nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)

    class_: Mapped["Class"] = relationship("Class", back_populates="students")
    grades: Mapped[list["Grade"]] = relationship("Grade", back_populates="student", cascade="all, delete-orphan")
